grin/test/AbstractInterpretation/ExtendedSyntax/SharingSpec.hs

```diff
@@ -24,7 +24,7 @@ spec :: Spec
 spec = describe "Sharing analysis" $ do
   it "has not changed for sum simple." $ do
     let result = calcSharedLocations testProgram
-    let expected = Set.fromList [0,1,4,5]
+    let expected = Set.fromList [0,1,2,4,5]
```

Comment:
As far as I remember, I used the result form Boquist's PhD. Why have been `2` bacame shared?

Reply:
Currently, this is how our implementation of sharing analysis work. It overapproximates the Boq PhD results. Location `2` became shared became our implementation was not correct, it did not collect the locations from non-linear variables containing node values (see 
34b7c93 ).

Reply:
Got it.